torch/csrc/jit/passes/concat_opt.cpp

```diff
@@ -510,5 +510,65 @@ void OptimizeConcat(const std::shared_ptr<Graph>& graph) {
   GRAPH_DUMP("After ConcatOpt", graph);
 }
 
+namespace {
+
+class VariadicCatUpdater {
+ public:
+  explicit VariadicCatUpdater(std::shared_ptr<Graph> graph)
+      : graph_(std::move(graph)) {}
+
+  void run() {
```

Comment:
You may want to run mutation removal pass first because that pass may remove some list mutation operations

Reply:
Same reason as above. Will add another higher level API to do both these until a fixed point.